### PR TITLE
3.3: initialize for minor release 3.3.0

### PR DESCRIPTION
**Pull request by ralfhandl**
- Initialize `src/oas.md` from `versions/3.2.0.md` on the `main` branch with empty history
- Bump version to 3.3.0
- Restore @karenetheridge's change from #5004
- Bump version in schema files
- Bump version in schema tests

This PR's head branch was created with `scripts/start-release.sh` from
- #5034

----
- [x] schema changes are included in this pull request

### CODE REVIEW

File: tests/schema/pass/valid_schema_types.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-openapi: 3.2.1
+openapi: 3.3.1
```
Comment (mikekistler) [inline tests/schema/pass/valid_schema_types.yaml:1]: Why is this file "3.3.1" but all the others are "3.3.0"?

Comment (ralfhandl) [reply tests/schema/pass/valid_schema_types.yaml:1]: This test case also tests a non-zero patch version number in the OpenAPI field, which should pass validation with a 3.3 schema, and it does.

Comment (mikekistler) [reply tests/schema/pass/valid_schema_types.yaml:1]: So it is acceptable to use non-existent patch version numbers? So "3.3.987" also validates? That seems odd.

But if that's the case, can we add a comment in the file to make that clear?

Comment (ralfhandl) [reply tests/schema/pass/valid_schema_types.yaml:1]: Our schemas traditionally allow any patch version, even ones that haven't been published yet:

https://github.com/OAI/OpenAPI-Specification/blob/de76eb4de13112080f55843c4a22ed32f3468fad/src/schemas/validation/schema.yaml#L4-L10

This test case was copied from `v3.2-dev`, here's the original, referencing a not-yet-published v3.2.1:

https://github.com/OAI/OpenAPI-Specification/blob/55816a2c384b9a5fdb88634f1015ff16a5d40a37/tests/schema/pass/valid_schema_types.yaml#L1

As stated in the PR description

> This PR's head branch was created with `scripts/start-release.sh` from
> - #5034

It is "proof of correctness" for #5034, and I hesitate to "taint" it with additional commits.

Can we improve this copied and auto-adjusted schema test case in a separate PR, or wait with adding to this PR until #5034 is merged?

Comment (ralfhandl) [reply tests/schema/pass/valid_schema_types.yaml:1]: To make sure it doesn't get forgotten:

- #5049
